SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -74,89 +74,184 @@ SpatialEventTracer::~SpatialEventTracer()
 	if (IsEnabled())
 	{
 		UE_LOG(LogSpatialEventTracer, Log, TEXT("Spatial event tracing disabled."));
-		Trace_EventTracer_Disable(EventTracer);
 		Trace_EventTracer_Destroy(EventTracer);
 	}
 }
 
-FString SpatialEventTracer::SpanIdToString(const Trace_SpanId& SpanId)
+FUserSpanId SpatialEventTracer::GDKSpanIdToUserSpanId(const FSpatialGDKSpanId& SpanId)
 {
-	FString HexStr;
-	for (int i = 0; i < TraceSpanIdLength; i++)
+	if (!SpanId.IsValid())
 	{
-		HexStr += FString::Printf(TEXT("%02x"), SpanId.data[i]);
+		return {};
 	}
-	return HexStr;
-}
 
-FUserSpanId SpatialEventTracer::SpanIdToUserSpanId(const Trace_SpanId& SpanId)
-{
 	FUserSpanId UserSpanId;
-	UserSpanId.Data.SetNum(TraceSpanIdLength);
-	const int32 Size = TraceSpanIdLength * sizeof(uint8);
-	FMemory::Memcpy(UserSpanId.Data.GetData(), SpanId.data, Size);
+	UserSpanId.Data.SetNum(TRACE_SPAN_ID_SIZE_BYTES);
+	const int32 Size = TRACE_SPAN_ID_SIZE_BYTES * sizeof(uint8);
+	FMemory::Memcpy(UserSpanId.Data.GetData(), SpanId.GetConstData(), Size);
```

Comment:
The size is already implied in the macro!
```suggestion
	FMemory::Memcpy(UserSpanId.Data.GetData(), SpanId.GetConstData(), TRACE_SPAN_ID_SIZE_BYTES);
```

Reply:
Hehe true.